### DISCUSSION TITLE
Can I rename a JS chunk for a page?

### DISCUSSION THREAD

**Discussion by tmaximini** [vercel/next.js#33178] [Help] [answered]
Hi,
I have a page called "Analytics" in my next.js app. By default, nextjs generates a JS chunk for each page and this chunk is called "analytics.js". The problem is that many ad blockers prevent this chunk from getting loaded. 
So I was hoping I could tell next to use a different chunk name for that specific page. Any idea on how to do that?
Thank you

**Comment by shuding** [comment, marked as answer]
You can rename the page file as, e.g. pages/foo.js, and then add a `rewrites` rule to map it to /analytics.

**Reply by tmaximini** [reply]
@shuding That works, but in this case it shows /foo in the address bar of the browser as well. I'd like to keep the url as /analytics if possible.

**Reply by optimistic-updt** [reply]
did you ever figure it out @tmaximini ? I have the exact same problem 🤣
